.github/workflows/check-links.yml

```diff
@@ -0,0 +1,49 @@
+name: Check links of the site
+on:
+  # TODO schedule and workflow_dispatch in the final implementation
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  check_links:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4 # access to the local action
+      - name: Build Setup
+        uses: ./.github/actions/build-setup
+        ## TODO duplicated with the build-pr-site action --> create a shared action
+        # only require if we build a part of the documentation
+      - name: Install antora dependencies for xref validation
+        run: npm install --save-dev antora@3.2.0-alpha.2
+      - name: Build site
+        run: |
+          ./build-preview-dev.bash --hide-edit-page-links true --force-production-navbar true --type links-check --component bonita --branch 2023.1
```

Comment:
⚠️ Currently, we only check a single bonita branch, we should check the whole site